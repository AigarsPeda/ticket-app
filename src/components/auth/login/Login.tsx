import React, { useState } from "react";
import { Link } from "react-router-dom";

import "../Auth.scss";

import { IUser, IError } from "../../../interfaces/interfaces";
import { validateInputs } from "../../../helpers/helpers";

import FormInput from "../../reusable/FormInput";
import Button from "../../reusable/Button";

const Login: React.FC = () => {
  const [user, setUser] = useState<IUser>({
    username: "",
    password: "",
  });

  const [error, setError] = useState<IError>({
    usernameError: "",
    passwordError: "",
  });

  const { username, password } = user;
  const { usernameError, passwordError } = error;

  const onLoginUser = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    const isValid = validateInputs(user, setError);

    if (isValid) {
      console.log(user);
    }
  };

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setUser((state) => ({
      ...state,
      [name]: value,
    }));
  };

  return (
    <div className="auth-wrapper">
      <div className="auth-inner">
        <form onSubmit={onLoginUser}>
          <h3>Sign In</h3>
          <div className="form-group">
            <FormInput
              type="text"
              name="username"
              label="Username"
              className="form-control"
              placeholder="Enter Username"
              value={username}
              error={usernameError}
              onChange={handleChange}
            />
          </div>
          <div className="form-group">
            <FormInput
              type="password"
              name="password"
              label="Password"
              className="form-control"
              placeholder="Enter Password"
              value={password}
              error={passwordError}
              onChange={handleChange}
            />
          </div>
          <Button
            type="submit"
            label="Sign Up"
            className="btn btn-primary btn-block"
            // handleClick={onRegisterUser}
          />
          <p className="forgot-password text-right">
            Not yet registered?
            <Link to={"/sign-up"}>Sign Up</Link>
          </p>
        </form>
      </div>
    </div>
  );
};

export default Login;
