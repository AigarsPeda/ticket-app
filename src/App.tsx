import React from "react";
import { Switch, Route } from "react-router-dom";

import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import store, { persistor } from "./redux/store";

import Login from "./components/auth/login/Login";
import Register from "./components/auth/register/Register";
import Dashboard from "./components/main/home/Dashboard";
import PrivateRoute from "./components/private/PrivateRoute";
import Navbar from "./components/layouts/Navbar";
// import NoMatch from "./components/noMatch/NoMatch";

class App extends React.PureComponent {
  render() {
    return (
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <Navbar />
          <div className="App container">
            <Switch>
              <PrivateRoute exact path="/dashboard" component={Dashboard} />
              <Route exact path="/" component={Login} />
              <Route exact path="/sign-in" component={Login} />
              <Route exact path="/sign-up" component={Register} />
              <Route component={Login} />
            </Switch>
          </div>
        </PersistGate>
      </Provider>
    );
  }
}

export default App;
