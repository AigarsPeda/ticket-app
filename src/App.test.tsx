import React from "react";
import ReactDOM from "react-dom";

import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import App from "./App";
import Login from "./components/auth/login/Login";

Enzyme.configure({ adapter: new Adapter() });

describe("<App />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
  });

  it("invalid path should redirect to Login", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/random"]}>
        <App />
      </MemoryRouter>
    );
    // expect(wrapper.find(LandingPage)).toHaveLength(0);
    expect(wrapper.find(Login)).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
