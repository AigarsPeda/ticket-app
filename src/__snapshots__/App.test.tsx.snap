// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App container"
>
  <div
    className="auth-wrapper"
    data-testid="loginComponent"
  >
    <div
      className="auth-inner"
    >
      <form
        onSubmit={[Function]}
      >
        <h3>
          Sign In
        </h3>
        <div
          className="form-group"
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            autoComplete="false"
            className="form-control"
            name="username"
            onChange={[Function]}
            placeholder="Enter Username"
            style={
              Object {
                "border": undefined,
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            autoComplete="false"
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            style={
              Object {
                "border": undefined,
              }
            }
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-primary btn-block"
          data-testid="button"
          type="submit"
        >
          Sign Up
        </button>
        <p
          className="forgot-password text-right"
        >
          Not yet registered?
          <a
            href="/sign-up"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
